### ISSUE TITLE
🐛 BUG: Conditionally rendering Markdown with remote content causes parse error

### ISSUE THREAD

**Issue by mattjennings** [snowpackjs/astro#1061]
### What package manager are you using?

pnpm

### What operating system are you using?

Mac

### Describe the Bug

When conditionally rendering Markdown with remote content inside of a JSX expression, astro will throw a parse error:

```
[parse-error]

 /sandbox/src/pages/index.astro:13:111

 𝘅 Expected ")" but found end of file

     11 |   <body>
     12 |       <div>
   > 13 |          {showMarkdown && <Markdown content="test" />}
        |                                                       ^
     14 |       </div>
     15 |   </body>
     16 | </html>

(node:61) UnhandledPromiseRejectionWarning: Error: Unable to compile expression
    at compileExpression (file:///sandbox/node_modules/astro/src/compiler/codegen/index.ts:256:28)
    at Object.enter (file:///sandbox/node_modules/astro/src/compiler/codegen/index.ts:606:24)
    at AsyncWalker.visit (file:///sandbox/node_modules/astro/node_modules/estree-walker/src/async.js:49:5)
    at AsyncWalker.visit (file:///sandbox/node_modules/astro/node_modules/estree-walker/src/async.js:86:6)
    at AsyncWalker.visit (file:///sandbox/node_modules/astro/node_modules/estree-walker/src/async.js:79:14)
    at AsyncWalker.visit (file:///sandbox/node_modules/astro/node_modules/estree-walker/src/async.js:79:14)
    at AsyncWalker.visit (file:///sandbox/node_modules/astro/node_modules/estree-walker/src/async.js:79:14)
    at AsyncWalker.visit (file:///sandbox/node_modules/astro/node_modules/estree-walker/src/async.js:79:14)
    at asyncWalk (file:///sandbox/node_modules/astro/node_modules/estree-walker/src/index.js:34:9)
```

Same error as #1060, so it might be related.

### Steps to Reproduce

Create a .astro file with the following content:

```astro
---
import { Markdown } from 'astro/components';

const showMarkdown = true
---

<html>
  <head>
    <title>Hello Astro!</title>
  </head>
  <body>    
      <div>
         {showMarkdown && <Markdown content="test" />}
      </div>
  </body>
</html>
```

### Link to Minimal Reproducible Example (Optional)

https://codesandbox.io/s/aged-mountain-e7ovf?file=/src/pages/index.astro

**Comment by jasikpark** [comment]
What happens if you wrap `<Markdown/>` in a `<Fragment/>`? Since that seems to be a solution they found... I think maybe the markdown component is returning multiple elements with no wrapping context, so that might be a "fix" though I think this is definitely a bug 🐛

**Comment by mattjennings** [comment]
Yup, wrapping in a fragment seems to fix this as well!

**Comment by drwpow** [comment]
#1059, #1060, and #1061 are all great issues, thank you! We may wait to solve these in the [new compiler](https://twitter.com/astrodotbuild/status/1426219847671681024?s=20) to make sure these are squashed for good. Either way, will keep an eye on these.

**Comment by natemoo-re** [comment]
Confirming that this is fixed in the `astro@next` preview! See [working reproduction](https://stackblitz.com/edit/github-x3dyfv?file=src%2Fpages%2Findex.astro).

Will put this in our milestone to be closed when that release is stable.

**Comment by natemoo-re** [comment]
Following up on my previous comment, the team has decided to close out issues that have been confirmed as fixed by [astro@0.21.0-next.0](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.0), [astro@0.21.0-next.1](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.1), or [astro@0.21.0-next.2](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.2). Our hope is that this will help the [v0.21 milestone](https://github.com/snowpackjs/astro/milestone/1) remain as actionable as possible.

To verify that this issue has been fixed, you may
- Open a new [`astro@next` project on StackBlitz](https://astro.new/minimal@next)
- Locally upgrade to the `astro@next` preview
    ```shell
    npm i astro@next
    # If using Framework renderers...
    npm i @astrojs/renderer-name@next
    ```

If you think we made a mistake, **please reopen this issue**!

If you run into any other problems with `astro@next`, please [open a new issue](https://github.com/snowpackjs/astro/issues/new?assignees=&labels=0.21&template=---01-bug-report.yml&title=%F0%9F%90%9B+BUG%3Aastro@next).
